src/sink.cpp

```diff
@@ -161,14 +165,15 @@ std::string Sink::formatted_log_plaintext_short(Log const& log, Channel const& c
     return ss.str();
 }
 
-std::string Sink::formatted_context_info(Log const& log,
-                                         Channel const& channel,
-                                         Logger::ContextInfo const& context_info) const
+std::string Sink::formatted_context_info(Log const& log, Channel const& channel, ContextInfo const& context_info, ContextInfo const& global_context_info) const
 {
     std::string context_info_str("context_info: ");
-    for (auto const& itr : context_info)
+    for (auto const& itr : {log.context_info(), context_info, global_context_info})
```

Comment:
Wouldn't this just create 3 copies even though they are marked `const`?

Reply:
why?
all are by reference?

Reply:
I guess you're right.